src/track/beatutils.cpp

```diff
@@ -43,17 +43,21 @@ QMap<int, double> BeatUtils::findTempoChanges(
     // Since we use the median as a guess first and last tempo
     // And it can not have values outside from tempoFrequency
     const double median = computeSampleMedian(sortedTempoList);
-    // The analyzer sometimes detect false beats that generate outliers
-    // values for the tempo so we use a median filter to remove them
-    MovingMedian filterTempo(5); // 5 is the lenght our window
+    // Forming a meter perception takes a few seconds, so we assume sections of consistent 
+    // metrical structure to be at least around 10s long. So we use a window of the double
+    // of that in our filtering.  
```

Comment:
Add a brief TODO comment here about your ideas to improve this further in the future which you mentioned on Zulip.

Reply:
done